source/elements/oneMKL/source/domains/spblas/data_types/create_dense_matrix.rst

```diff
@@ -0,0 +1,192 @@
+
+In the case of buffers, the reference count of the provided buffer is
+incremented which extends the lifetime of the underlying buffer until the
+``dmhandle`` is destroyed with ``destroy_dense_matrix`` or the data is reset
+with ``set_dense_matrix_data``.
+
+In the case of USM, the object does not take ownership of the data.
+
+See :ref:`onemkl_sparse_dense_matrix_handle`.
+
+.. _onemkl_sparse_create_dense_matrix_buffer:
+
+create_dense_matrix (Buffer version)
+------------------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       template <typename dataType>
+       void create_dense_matrix (sycl::queue                                &queue,
+                                 oneapi::mkl::sparse::dense_matrix_handle_t *p_dmhandle,
+                                 std::int64_t                               num_rows,
+                                 std::int64_t                               num_cols,
+                                 std::int64_t                               ld,
+                                 layout                                     dense_layout,
+                                 sycl::buffer<dataType, 1>                  &val);
+
+   }
+
+.. container:: section
+
+   .. rubric:: Template parameters
+
+   dataType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`.
+
+.. container:: section
+
+   .. rubric:: Input parameters
+
+   queue
+      The SYCL command queue which will be used for SYCL kernels execution.
+
+   p_dmhandle
+      The address of the ``p_dmhandle`` object to be initialized. Must only be
+      called on an uninitialized ``dense_matrix_handle_t`` object.
+
+   num_rows
+      Number of rows of the provided data ``val``. Must be at least 0.
+
+   num_cols
+      Number of columns of the provided data ``val``. Must be at least 0.
+
+   ld
+      Leading dimension of the provided data ``val``. Must be at least
+      ``num_rows`` if column major layout is used or at least ``num_cols`` if
+      row major layout is used.
+
+   dense_layout
+      Specify whether the data uses row major or column major.
+
+   val
+      Buffer of length ``ld*num_cols`` if column major is used or
+      ``ld*num_rows`` if row major is used. Holds the data to initialize
+      ``p_dmhandle`` with.
+
+.. container:: section
+
+   .. rubric:: Output parameters
+
+   p_dmhandle
+      On return, the address is updated to point to a newly allocated and
+      initialized ``dense_matrix_handle_t`` object that can be filled and used
+      to perform sparse BLAS operations.
+
+.. container:: section
+
+   .. rubric:: Throws
+
+   This function shall throw the following exceptions if the associated
+   condition is detected. An implementation may throw additional
+   implementation-specific exception(s) in case of error conditions not covered
+   here.
+
+   | :ref:`oneapi::mkl::device_bad_alloc<onemkl_exception_device_bad_alloc>`
```

Comment:
I wonder if `device_bad_alloc` makes sense for this API given it's not supposed to allocate anything on the device. @spencerpatty, should we remove this particular exception from this page?

Reply:
I agree so I have removed this exception in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/4db05aa0dc4b077fc6005f84682de590810718f5

Reply:
yes good point, only host side allocations so not a potential exception.